File: backend/models/experiance.model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

//connection
require('../database/mongoose')

// creating the collection
const ExperianceSchema = mongoose.Schema({
    start_year : { type: String, required: [true, "Please fill this field"] },
    end_year : { type: String, required: [true, "Please fill this field"] },
    company : { type: String, required: [true, "Please fill this field"] },
    description : { type: String }
});

// to create data objects based on the schema we need model
module.exports = Experiance = mongoose.model('Experiance',ExperianceSchema);

File: frontend/lib/models/Experiance.model.dart
import 'User.model.dart';

class Experiance {
  String id;
  String startDate;
  String endDate;
  String company;
  String description;
  String user;

  Experiance({
    required this.id,
    required this.startDate,
    required this.endDate,
    required this.company,
    required this.description,
    required this.user
  });

  factory Experiance.fromJson(Map<String, dynamic> json) {
    return Experiance(
        id: json['result']['_id'],
        startDate: json['result']['start_year'],
        endDate: json['result']['end_year'],
        company: json['result']['company'],
        description: json['result']['description'],
        user: json['result']['user']
    );
  }
}
